engines/crab/ui/FileMenu.h

```diff
@@ -0,0 +1,255 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/*
+ * This code is based on the CRAB engine
+ *
+ * Copyright (c) Arvind Raja Yadav
+ *
+ * Licensed under MIT
+ *
+ */
+
+#ifndef CRAB_FILEMENU_H
+#define CRAB_FILEMENU_H
+
+#include "common/savefile.h"
+#include "crab/crab.h"
+#include "crab/ui/FileData.h"
+#include "crab/ui/ImageData.h"
+#include "crab/ui/PageMenu.h"
+#include "crab/ui/TextData.h"
+
+namespace Crab {
+
+namespace pyrodactyl {
+namespace ui {
+// Used for menus that are responsible for reading multiple files from disk
+template<typename FileType>
+class FileMenu {
+protected:
+	// The background of the menu
+	ImageData _bg;
+
+	// The collection of buttons
+	PageButtonMenu _menu;
+
+	// The final filename that is selected
+	Common::String _selected;
+
+	// The extension and directory used by this menu
+	Common::String _extension, _directory;
+
+	// The save information for each slot
+	Common::Array<FileType> _slotInfo;
+	TextData tdB[DATA_BUTTON_TOTAL];
+
+	// The titles for loc_name, difficulty, time_played and player_name
+	HoverInfo hov[DATA_HOVER_TOTAL];
+	TextData tdH[DATA_HOVER_TOTAL];
+
+	// The preview picture details
+	struct {
+		// We load only the current preview image instead of all of them
+		pyrodactyl::image::Image _preview;
+
+		// Fallback path if there is no preview image or if we fail to load it
+		Common::String _noPreviewPath;
+
+		// Position of image
+		Element _pos;
+
+		// Is the image loaded
+		bool _loaded;
+	} _img;
+
+	// Are we hovering over a button right now?
+	bool _hover;
+
+	// The previously hover button
+	int _prevHover;
+
+public:
+	FileMenu() {
+		_img._loaded = false;
+		_hover = false;
+		_prevHover = -1;
+	}
+
+	~FileMenu() {
+		if (_img._loaded)
+			_img._preview.deleteImage();
+	}
+	void reset() {
+		if (_img._loaded)
+			_img._preview.deleteImage();
+		_img._loaded = false;
+		_hover = false;
+	}
+
+	Common::String selectedPath() {
+		return _selected;
+	}
+
+	void selectedPath(const Common::String &val) {
+		_selected = val;
+	}
+
+	void scanDir() {
+		Common::String res = "CRAB_*";
```

Comment:
You mentioned that the savefiles are named UNREST.###. Is this a leftover, then?